pkg/resource/table/hooks.go

```diff
@@ -403,11 +443,14 @@ func customPreCompare(
 	a *resource,
 	b *resource,
 ) {
-
```

Comment:
The error we're seeing in the logs "can" be solved doing something like: https://github.com/aws-controllers-k8s/dynamodb-controller/blob/main/pkg/resource/table/hooks.go#L494-L498 - this tricks the delta function in believing that "nil" struct is equivalent to `spec.x = &{y: &false}`